packages/inventory/src/Stocktake/DetailView/columns.ts

```diff
@@ -2,42 +2,166 @@ import {
+            if ('lines' in row) {
+              const { lines } = row;
+              return ifTheSameElseDefault(lines, 'batch', '[multiple]') ?? '';
+            } else {
+              return row.batch ?? '';
+            }
+          },
+          accessor: ({ rowData }) => {
+            if ('lines' in rowData) {
+              const { lines } = rowData;
+              return ifTheSameElseDefault(lines, 'batch', '[multiple]');
+            } else {
+              return rowData.batch;
+            }
+          },
+        },
+      ],
+      [
+        'expiryDate',
+        {
+          getSortValue: row => {
+            if ('lines' in row) {
+              const { lines } = row;
+              const expiryDate =
+                ifTheSameElseDefault(lines, 'expiryDate', null) ?? '';
+              return (expiryDate && formatExpiryDate(expiryDate)) || '';
+            } else {
+              return formatExpiryDate(row.expiryDate) ?? '';
+            }
+          },
+          accessor: ({ rowData }) => {
+            if ('lines' in rowData) {
+              const { lines } = rowData;
+              const expiryDate = ifTheSameElseDefault(
+                lines,
+                'expiryDate',
+                null
+              );
+              return formatExpiryDate(expiryDate);
+            } else {
+              return formatExpiryDate(rowData.expiryDate);
+            }
+          },
+        },
+      ],
+
       {
         key: 'countedNumPacks',
         label: 'label.counted-num-of-packs',
         width: 150,
         align: ColumnAlign.Right,
+        getSortValue: row => {
+          if ('lines' in row) {
+            const { lines } = row;
+            return ifTheSameElseDefault(lines, 'countedNumPacks', '') ?? '';
+          } else {
+            return row.countedNumPacks ?? '';
+          }
+        },
+        accessor: ({ rowData }) => {
+          if ('lines' in rowData) {
+            const { lines } = rowData;
+            return ifTheSameElseDefault(lines, 'countedNumPacks', '');
+          } else {
+            return rowData.countedNumPacks;
+          }
+        },
       },
+
       {
         key: 'snapshotNumPacks',
         label: 'label.snapshot-num-of-packs',
         align: ColumnAlign.Right,
+        getSortValue: row => {
+          if ('lines' in row) {
+            const { lines } = row;
+            return ifTheSameElseDefault(lines, 'snapshotNumPacks', '') ?? '';
+          } else {
+            return row.snapshotNumPacks ?? '';
+          }
+        },
+        accessor: ({ rowData }) => {
+          if ('lines' in rowData) {
+            const { lines } = rowData;
+            return ifTheSameElseDefault(lines, 'snapshotNumPacks', '');
+          } else {
+            return rowData.snapshotNumPacks;
+          }
+        },
       },
 
       GenericColumnKey.Selection,
     ],
-    { onChangeSortBy, sortBy },
-    [sortBy]
+    { sortBy, onChangeSortBy },
+    [sortBy, onChangeSortBy]
```

Comment:
TODO: I'm not sure if these columns are the right ones we want - so need to double check that

Reply:
Yes to all! Thanks